File: examples/CBuildr3/ExDbCbu.cpp
//---------------------------------------------------------------------------
#include <vcl\vcl.h>
#pragma hdrstop

#include "ExDbCbu.h"
//---------------------------------------------------------------------------
#pragma link "OvcDbTbl"
#pragma link "OvcTCmmn"
#pragma link "OvcBase"
#pragma link "OvcTCCBx"
#pragma link "OvcTCStr"
#pragma link "OvcTCell"
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
  : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TForm1::FormCreate(TObject *Sender)
{
  Table2->First();
  while (!Table2->Eof) {
    OvcTCComboBox1->Items->Add(Table2->Fields[0]->AsString);
    Table2->Next();
  }
}
//---------------------------------------------------------------------------

File: examples/Delphi/Table/DB CB/EXDBCBU.pas
{$J+} {Writable constants}

unit Exdbcbu;

interface

uses
  WinProcs, WinTypes, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  OvcTCmmn, OvcTCell, OvcTCStr, OvcTCCBx, OvcBase, OvcDbTbl, DB, DBTables,
  ExtCtrls, DBCtrls, OvcTbCls, StdCtrls, OvcTCBEF, OvcTCSim;

type
  TForm1 = class(TForm)
    DataSource1: TDataSource;
    Table1: TTable;
    OvcDbTable1: TOvcDbTable;
    OvcController1: TOvcController;
    OvcTCComboBox1: TOvcTCComboBox;
    DataSource2: TDataSource;
    Table2: TTable;
    Table1NAME: TStringField;
    Table1SIZE: TSmallintField;
    DBNavigator1: TDBNavigator;
    Memo1: TMemo;
    OvcTCSimpleField1: TOvcTCSimpleField;
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

{$R *.DFM}

procedure TForm1.FormCreate(Sender: TObject);
begin
  Table2.First;
  while not (Table2.EOF) do begin
    OvcTCComboBox1.Items.Add(Table2.Fields[0].AsString);
    Table2.Next;
  end;
end;

end.
